test/integration/multicluster/install_test.go

```diff
@@ -77,47 +135,234 @@ func TestMulticlusterResourcesPostInstall(t *testing.T) {
 	testutil.TestResourcesPostInstall(TestHelper.GetMulticlusterNamespace(), multiclusterSvcs, testutil.MulticlusterDeployReplicas, TestHelper, t)
 }
 
-// TestInstallViz will install the viz extension, needed to verify whether the
-// gateway probe succeeded.
-// TODO (matei): can the dependency on viz be removed?
-func TestInstallViz(t *testing.T) {
-	cmd := []string{
-		"viz",
-		"install",
-		"--set", fmt.Sprintf("namespace=%s", TestHelper.GetVizNamespace()),
+func TestLinkClusters(t *testing.T) {
+	linkName := "target"
+	// Get gateway IP from target cluster
+	lbCmd := []string{
+		"--context=" + contexts["tgt"],
+		"get", "svc",
+		"-n", "kube-system", "traefik",
+		"-o", "go-template={{ (index .status.loadBalancer.ingress 0).ip }}",
 	}
+	lbIP, err := TestHelper.KubectlWithContext("", contexts["tgt"], lbCmd...)
 
-	out, err := TestHelper.LinkerdRun(cmd...)
 	if err != nil {
-		testutil.AnnotatedFatal(t, "'linkerd viz install' command failed", err)
+		testutil.AnnotatedFatalf(t, "'kubectl get' command failed",
+			"'kubectl get' command failed\n%s", lbIP)
+	}
+	linkCmd := []string{
+		"--context=" + contexts["tgt"],
+		"multicluster", "link",
+		"--log-level", "debug",
+		"--api-server-address", fmt.Sprintf("https://%s:6443", lbIP),
+		"--cluster-name", linkName, "--set", "enableHeadlessServices=true",
 	}
 
-	out, err = TestHelper.KubectlApplyWithArgs(out)
+	// Create link in target context
+	out, err := TestHelper.LinkerdRun(linkCmd...)
+	if err != nil {
+		testutil.AnnotatedFatalf(t, "'linkerd multicluster link' command failed", "'linkerd multicluster link' command failed: %s\n%s", out, err)
+	}
+
+	// Apply Link in source
+	out, err = TestHelper.KubectlApplyWithContext(out, contexts["src"], "-f", "-")
 	if err != nil {
 		testutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
 			"'kubectl apply' command failed\n%s", out)
 	}
 
-	TestHelper.WaitRollout(t, testutil.LinkerdVizDeployReplicas)
+}
+
+// TestInstallViz will install the viz extension, needed to verify whether the
+// gateway probe succeeded.
+// TODO (matei): can the dependency on viz be removed?
+func TestInstallViz(t *testing.T) {
+	for _, ctx := range contexts {
+		cmd := []string{
+			"--context=" + ctx,
+			"viz",
+			"install",
+			"--set", fmt.Sprintf("namespace=%s", TestHelper.GetVizNamespace()),
+		}
+
+		out, err := TestHelper.LinkerdRun(cmd...)
+		if err != nil {
+			testutil.AnnotatedFatal(t, "'linkerd viz install' command failed", err)
+		}
+
+		out, err = TestHelper.KubectlApplyWithContext(out, ctx, "-f", "-")
+		if err != nil {
+			testutil.AnnotatedFatalf(t, "'kubectl apply' command failed",
+				"'kubectl apply' command failed\n%s", out)
+		}
+
+		TestHelper.WaitRolloutWithContext(t, testutil.LinkerdVizDeployReplicas, ctx)
```

Comment:
I'll loop again and block on viz readiness. Since we're allowing it to be installed in parallel, my feeling is we won't block for a long time in worst case scenarios; if one is ready, it's likely in the next loop iteration we won't block for as long as we did in the first iteration (if that makes sense).